microsoft-edge/webview2/how-to/hostobject.md

```diff
@@ -0,0 +1,262 @@
+---
+title: Using Host Objects in WebView2
+description: How to pass host objects to JavaScript using the AddHostObjectToScript API for WebView2 apps.
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.prod: microsoft-edge
+ms.technology: webview
+ms.date: 2/25/2022
+---
+# Using Host Objects in WebView2
+
+WebView2 enables applications to bridge the gap between the web and native sides of an application by enabling an object to be passed to the web. Such objects are defined in the native code and often called *host objects*. They can be projected into JavaScript using the WebView2 `AddHostObjectToScript` API, as described in this document.
+
+Why use `AddHostObjectToScript`?
+
+  * When developing a WebView2 app, you may encounter a native object whose methods or properties you find useful. You might want to trigger these native object methods from web-side code, or as a result of user interaction on the web side of your app. In addition, you might not want to re-implement your native objects' methods in your web-side code.  The `AddHostObjectToScript` API enables re-use of native-side code by web-side code. 
+
+  * For example, there might be a native webcam API, which would require re-writing a large amount of code on the web side. Having the ability to call the native object's methods is quicker and more efficient than re-coding the object's methods on the web side of your app. In this case, your native-side code can pass the object to your app's web-side, JavaScript code, so that your JavaScript code can reuse the native API's methods.
+
+Scenarios that may benefit from using host objects in script:
+
+  * There is a keyboard API, and you want to call the `keyboardObject.showKeyboard` function from the web side.
+
+  * JavaScript is sandboxed, limiting its ability on the native side. For example, if you need to access a file on the native side, you must use the native file system. If you have a native object exposed to JavaScript via `AddHostObjectToScript`, you can use it to manipulate files on the native file system.
+
+This article uses the WebView2 Win32 sample app to demonstrate some practical applications of `AddHostObjectToScript`. For more information about how to embed web content into native applications, see [Embed web content into native applications](/microsoft-edge/webview2/how-to/communicate-btwn-web-native).
```

Comment:
Should this be a link?

Reply:
It is. It works in the rendered version. Is it not working for you?

Reply:
I meant the reference to our WebView2 Win32 Sample App. This is the repo: [WebView2APISample](https://github.com/MicrosoftEdge/WebView2Samples/tree/master/SampleApps/WebView2APISample)

Reply:
yes, fixing